src/nunit.analyzers/SameAsIncompatibleTypes/SameAsIncompatibleTypesAnalyzer.cs

```diff
@@ -29,17 +29,17 @@ public class SameAsIncompatibleTypesAnalyzer : BaseAssertionAnalyzer
             var cancellationToken = context.CancellationToken;
             var semanticModel = context.SemanticModel;
 
-            if (!AssertExpressionHelper.TryGetActualAndConstraintExpressions(assertExpression,
+            if (!AssertExpressionHelper.TryGetActualAndConstraintExpressions(assertExpression, semanticModel,
                 out var actualExpression, out var constraintExpression))
             {
                 return;
             }
 
-            var sameAsExpectedExpressions = AssertExpressionHelper
-                .GetExpectedArguments(constraintExpression, semanticModel, cancellationToken)
-                .Where(ex => ex.constraintMethod.Name == NunitFrameworkConstants.NameOfIsSameAs
-                    && ex.constraintMethod.ReturnType.GetFullMetadataName() == NunitFrameworkConstants.FullNameOfSameAsConstraint)
-                .Select(ex => ex.expectedArgument)
+            var sameAsExpectedExpressions = constraintExpression.ConstraintParts
+                .Where(part => part.GetConstraintName() == NunitFrameworkConstants.NameOfIsSameAs
+                    && part.GetConstraintMethod()?.ReturnType?.GetFullMetadataName() == NunitFrameworkConstants.FullNameOfSameAsConstraint)
```

Comment:
Well, technically you can have some custom constraint expression, where SameAs is not a method, but a property. 
Have serious doubts that it's going to happen, yeah :)

Still, I would leave as is - if we start using Nullable Ref Types annotations - it will show warning here.


Probably can safely remove `?.` for ReturnType though - it can't ever be null for IMethodSymbol

Reply:
Let's just keep them. At some point I want to take a look at Nullable Ref Types annotations. I was mostly curious if anything special had triggered the adding of `?`.